[quant] Add quantized levels visualization

asl3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79499
* #79530
* __->__ #79198

### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf).

### Test Plan:
Run unit tests with:`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
Sample generated plots are displayed below for the corresponding test cases and hard-coded parameters in `test_nonuniform_observer.py`. Three visualizations were generated for each of the non-empty observers being tested. 

Parameters are defined as in the paper, where:
 * b: total number of bits across all terms
 * k: base bitwidth
 * n: number of additive terms
 
**Test case 2:** b = 4, k = 2, n = 2, signed = False
<img width="599" alt="Screen Shot 2022-06-15 at 10 28 03 PM" src="https://user-images.githubusercontent.com/68875504/173978220-44e088fc-09a8-401c-a7c5-b6c11f5e429a.png">
**Test case 3:** b = 6, k = 2, n = 3, signed = False
<img width="582" alt="Screen Shot 2022-06-15 at 10 28 35 PM" src="https://user-images.githubusercontent.com/68875504/173978261-888b814d-d5c5-4d7c-b590-60be795a5f80.png">
**Test case 4:** b = 4, k = 2, n = 2, signed = True
<img width="596" alt="Screen Shot 2022-06-15 at 10 28 20 PM" src="https://user-images.githubusercontent.com/68875504/173978241-3c423f22-9751-4c49-814b-b61c4e6f4efb.png">

asl3 committed:
[quant] Add quantized levels visualization

[ghstack-poisoned]

vadimkantorov commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -138,3 +138,5 @@ def forward(self, x_orig):
         self.min_val.copy_(min_val)
```
`min_val` is declared as `min_val=torch.Tensor,`. should it instead be declared as `min_val: torch.Tensor,`?

vadimkantorov replied:
also probably if b=0, k=0 (or other reasonable defaults) could be put into the base class constructor defaults, ApOTObserver can do without overriding the constructor at all

asl3 committed:
Update on "[quant] Add quantized levels visualization"




[ghstack-poisoned]

dzdang commented:
I feel we don't need to put the images in the repo

vadimkantorov commented:
they will still be added into the docs?

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested. The three visualizations are included below for reference:

### Test 2: 
(b=4, k=2, n=2, signed=False)
<img width="521" alt="Screen Shot 2022-06-09 at 4 55 13 PM" src="https://user-images.githubusercontent.com/68875504/172942755-9afad5c1-0188-4bdc-8325-5645b5303f2b.png">

### Test 3: 
(b=6, k=2, n=3, signed=False)
<img width="518" alt="Screen Shot 2022-06-09 at 4 56 09 PM" src="https://user-images.githubusercontent.com/68875504/172942896-673f128b-8723-4579-a05f-52385a7b3883.png">

### Test 4:
(b=4, k=2, n=2, signed=True)
<img width="538" alt="Screen Shot 2022-06-09 at 4 56 41 PM" src="https://user-images.githubusercontent.com/68875504/172942972-7c66b8b6-5307-4e16-8539-1c65e6c0f48d.png">

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

HDCharles commented on test/quantization/core/experimental/test_nonuniform_observer.py:
```diff
@@ -2,8 +2,24 @@
+import torch
 
 class TestNonUniformObserver(unittest.TestCase):
+    def quant_levels_visualization(self, obs, obs_result, filename):
```
nit: this would normally be a private method since its not an actual test, prepend with '_'

HDCharles reviewed (approved):
looks good other than some small nits

jerryzh168 commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -111,20 +113,39 @@ def _calculate_qparams(
-    def _calculate_qparams(self, min_val, max_val, signed):
-        return super(APoTObserver, self)._calculate_qparams(min_val, max_val, signed)
+    def _calculate_qparams(self, signed, min_val, max_val):
+        return super(APoTObserver, self)._calculate_qparams(signed, min_val, max_val)
```
why there is a standard way for NonUniformQuantizationObserverBase to calculate qprams?

asl3 replied:
yep this should be APoT calculate qparams

jerryzh168 commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -48,11 +49,12 @@ def __init__(
-            signed: bool) -> Tuple[float, torch.Tensor, torch.Tensor]:
+        signed: bool,
+        min_val=None,
+            max_val=None) -> Tuple[float, torch.Tensor, torch.Tensor]:
```
seems like indentation is messed up here

jerryzh168 commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -111,20 +113,39 @@ def _calculate_qparams(
 
         return (self.gamma, quantization_levels, level_indices)
 
+    def float_to_apot(self, x, levels, indices):
```
looks like this is quantization from float to apot int value, why are we attaching this to observer? I think it's better to define a standalone function for it

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

dzdang commented:
plz remove the plots from the PR before landing

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

dzdang reviewed (changes requested):
I think you might have rebased incorrectly? I don't think this PR should touch `test/quantization/core/experimental/test_nonuniform_observer.py`
those changes seem relevant for a later PR?

dzdang commented on test/quantization/core/experimental/test_nonuniform_observer.py:
```diff
@@ -2,8 +2,24 @@
+
+        plt.plot(xs, ys)
+        plt.title("APoT Quantization Plot")
+        plt.xlabel("Float")
```
@asl3  also this comment

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf) Three visualizations were generated for each of the non-empty obervers being tested.

### Test Plan:
`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
The generated plots are saved in the directory `pytorch/test/quantization/core/experimental/plots`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantized levels visualization"


### Summary:
This PR adds functionality to display visualization of the quantization levels calculated in the tests for non-uniform observers. The visualization follows the example shown in Figure 2 of the APoT paper (https://arxiv.org/pdf/1909.13144.pdf).

### Test Plan:
Run unit tests with:`python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py`
Sample generated plots are displayed below for the corresponding test cases and hard-coded parameters in `test_nonuniform_observer.py`. Three visualizations were generated for each of the non-empty observers being tested. 

Parameters are defined as in the paper, where:
 * b: total number of bits across all terms
 * k: base bitwidth
 * n: number of additive terms
 
**Test case 4:** b = 4, k = 2, n = 2, signed = False
<img width="599" alt="Screen Shot 2022-06-15 at 10 28 03 PM" src="https://user-images.githubusercontent.com/68875504/173978220-44e088fc-09a8-401c-a7c5-b6c11f5e429a.png">
**Test case 3:** b = 6, k = 2, n = 3, signed = False
<img width="582" alt="Screen Shot 2022-06-15 at 10 28 35 PM" src="https://user-images.githubusercontent.com/68875504/173978261-888b814d-d5c5-4d7c-b590-60be795a5f80.png">
**Test case 4:** b = 4, k = 2, n = 2, signed = True
<img width="596" alt="Screen Shot 2022-06-15 at 10 28 20 PM" src="https://user-images.githubusercontent.com/68875504/173978241-3c423f22-9751-4c49-814b-b61c4e6f4efb.png">



[ghstack-poisoned]

asl3 commented:
@pytorchbot merge -g
